framework/sandstone_tests.cpp

```diff
@@ -83,37 +83,47 @@ SandstoneTestSet::SandstoneTestSet(struct test_set_cfg cfg, bool all_tests, bool
     }
 };
 
+struct test_cfg_info SandstoneTestSet::enable(struct test *t) {
+    struct test_cfg_info ti;
+    if (!test_map.contains(t->id)) {
+        ti.st = TEST_ENABLED;
+        ti.test = t;
+        test_map[t->id] = ti;
+    } else {
+        ti = test_map[t->id];
+        ti.st = TEST_ENABLED;
+    }
```

Comment:
Unnecessary double map lookup. You always end up with the test in enabled state. If you ensure that `test_cfg_info` always initialises its members, you can reduce this to:
```c++
    test_cfg_info &ti = test_map[t->id];
    ti.status = test_cfg_info::enabled;
    ti.test = t;
    return ti;
```